test/e2e/main_test.go

```diff
@@ -116,13 +124,17 @@ func TestMain(m *testing.M) {
 				return ctx, err
 			}
 		}
-		relativeInstallDirectory := "../../install"
-		if cloudAPIAdaptor, err = pv.NewCloudAPIAdaptor(cloudProvider, relativeInstallDirectory); err != nil {
-			return ctx, err
-		}
-		log.Info("Deploy the Cloud API Adaptor")
-		if err = cloudAPIAdaptor.Deploy(ctx, cfg, provisioner.GetProperties(ctx, cfg)); err != nil {
-			return ctx, err
+
+		if shoudlInstallCAA {
+			log.Info("Install Cloud API Adaptor")
+			relativeInstallDirectory := "../../install"
+			if cloudAPIAdaptor, err = pv.NewCloudAPIAdaptor(cloudProvider, relativeInstallDirectory); err != nil {
```

Comment:
Even without installing CAA, don't we need to initialize the global `cloudAPIAdaptor` used elsewhere?

Reply:
The global var of cloudAPIAdaptor is only used in main_test.go to install CAA and delete CAA.

Reply:
To avoid confusion and problem, it's better to change the scope from global to local as this might be used by mistake somewhere else.